Updated to Python 3.6

mathematicalmichael opened the pull request:
tests pass, only matplotlib warnings remain. Backwards compatible with Python 2.7. 
Most of the fixes had to do with the default type being Float instead of Int.

mpilosov committed:
deprecation warnings unrelated to matplotlib silenced in py2.7

mpilosov committed:
fixed importing errors.

mpilosov committed:
all tests pass.

mpilosov committed:
README.

mathematicalmichael commented:
Follow-Up PR potential commits:
Re Travis CI... Is it as simple as adding `- 3.6` into line 7 of `.travis.yml`? 
Also, I had to `pip install PyQt5` - should this be added to `setup.py`?
Documentation updates?

mpilosov committed:
updated notebooks

mathematicalmichael committed:
plotting warnings addressed, notebooks updated.

mathematicalmichael committed:
notebooks validated with both versions.

mathematicalmichael committed:
travis update.

jordanrhall commented:
Thanks, Michael! I've been using this for a few weeks now. I can confirm that the "basic" tutorial (the one with the wing function) is running perfectly for me on a Windows machine running python 3.6. I also converted [some of my own personal research notebooks](https://github.com/jordanrhall/personalresearch/tree/master/Notebooks/active_subspaces_py3) that were using the old active subspace library to py3.6 using this package, and again, those ran perfectly on Windows. I can also confirm that one of my notebooks runs on Mac, py3.6 as well. If I'm not mistaken, that means we have success in py3.6 for Windows, Mac, and Linux.

mathematicalmichael commented:
@paulcon is there anything I'm missing from this PR that is preventing it from merge?

shahrozkhan66 commented:
@jordanrhall and @mathematicalmichael would mind if I ask what could be the reason I am getting this error AttributeError: module 'active_subspaces' has no attribute 'utils'.
I am pretty new to python and I have tried many things but I couldn't solve this error. I would really appreciate if any help regarding this. 
Bundle of Thanks!

mathematicalmichael commented:
@shahrozkhan66 how are you installing it and what code are you running to produce that output?

jordanrhall commented:
Hello,

This error generally means that your "active_subspaces" folder isn't saved in a place where your Python kernel can access it. An easy fix will (probably) be simply copying/pasting the entire active_subspaces folder into the working directory that contains the notebook/script you are trying to use the ac commands. You could also just clone a fresh copy of active_subspaces into that working directory.

Best, Jordan
________________________________
From: Michael Pilosov <notifications@github.com>
Sent: Tuesday, April 2, 2019 10:07 AM
To: paulcon/active_subspaces
Cc: Hall, Jordan R; Mention
Subject: Re: [paulcon/active_subspaces] Updated to Python 3.6 (#52)


@shahrozkhan66<https://github.com/shahrozkhan66> how are you installing it and what code are you running to produce that output?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/paulcon/active_subspaces/pull/52#issuecomment-479070611>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AUM2TVFW6OKi4osOXwuUG_DzfvEkGwn4ks5vc4AsgaJpZM4aCahJ>.

mathematicalmichael commented:
ah, in that case, running `python setup.py install` in the directory you cloned from github (either this master - which is incompatible with python 3), or the repo this pull request stems from (which is python 2 and 3 compatible). once you run that inside the folder, all the modules will be available no matter where you are. 

Try running your code in the environment that spawns up here: [![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/v2/gh/mathematicalmichael/active_subspaces/master)

mtezzele commented:
Is there anybody able to merge this PR? It would be of great interest the python 3 compatibility.

mathematicalmichael commented:
Yeah I’m not sure what the story is. This is backwards compatible with Python 2, which is reaching end of life in a couple months anyway. Fortunately you can still use it, just clone my fork and install from source. Been a while since I’ve done this so let me know if anything needs addressing.

Commit:
Update .travis.yml

Test on 3.7 and 3.8, quotes unnecessary

mathematicalmichael committed:
Merge pull request #2 from jordanrhall/patch-1

Update .travis.yml

mathematicalmichael commented:
looks like scipy 2.0 (default install) no longer supports some of the modules imported by this library. Another PR is open for fixing this by version-locking scipy to 1.2.2 and will be added on to here soon.

mathematicalmichael commented:
updating installation steps. conda on python 3.8 installs matplotlib 3.0, which appears to be an issue.

mathematicalmichael committed:
Update .travis.yml

trying a few things, including switching to `nosetests` syntax

mathematicalmichael committed:
Update setup.py

switch to pytest

mathematicalmichael committed:
Update .travis.yml

remove pytest install in hopes that pip catches it.

mathematicalmichael committed:
version lock scipy

scipy doesn't work for 2.0

mathematicalmichael committed:
Update .travis.yml

pytest and agg backend

mathematicalmichael committed:
Update .travis.yml

back to nosetests

mathematicalmichael committed:
Update setup.py

back to nosetests

mathematicalmichael committed:
Update .travis.yml

looks like not having conda install pip was an issue (the pip that was being called was the "wrong one")
Let's try to do this _without_ conda. Travis has its own system python versions cached anyway, so it should expedite tests a bit too.

mathematicalmichael committed:
Merge pull request #3 from mathematicalmichael/travis

Update CI pipeline
